### DISCUSSION TITLE
Upgrade (?) causes Floccus to see all bookmarks as deleted / Bookmark changed to Adobe Acrobat

### DISCUSSION THREAD

**Discussion by Sar6e** [nextcloud/bookmarks#2223] [Q&A] [answered]
I noticed this morning all my floccus extensions are reporting an error.  "E029: Failsafe: The current sync run would delete 93% of your bookmarks".  I had a quick look at the app in Nextcloud and all my bookmarks seem to be there, but I notice things look different, so I'm assuming the app in Nextcloud has been updated.

I'm guessing the upgrade process has made floccus see all the bookmarks as 'new' so it is trying to delete all the existing bookmarks and create new ones.  Does this sound about right?

I pulled the pin on one browser and overrode the fail safe.  Everything seems OK so no harm, no foul.  With one exception!

I had a bookmark in the root folder pointing to https://social.linux.pizza/home.  Looking at the Nextcloud app, this bookmark has been replaced with one pointing to https://acrobat.adobe.com/?x_api_client_id=bookmark&x_api_client_location=Acrobat

That is certainly a mystery...

**Comment by marcelklehr** [comment]
> I'm guessing the upgrade process has made floccus see all the bookmarks as 'new' so it is trying to delete all the existing bookmarks and create new ones. Does this sound about right?

This would be suprising to me. The app update shouldn't cause this.

Do you have logs of the faulty sync run?

**Comment by Sar6e** [comment]
I've just attempted a manual sync and received the same error.  I've attached the redacted log.  Is this the log file you need?

```
E029: Failsafe: The current sync run would delete 93% of your bookmarks. Refusing to execute. Disable this failsafe in the account settings if you want to proceed anyway. | Last synchronized: 1 day ago 
```
[floccus-5.2.7-2024-09-18-redacted.log](https://github.com/user-attachments/files/17036574/floccus-5.2.7-2024-09-18-redacted.log)

**Reply by marcelklehr** [reply, marked as answer]
Yes, thank you for the log file. It's quite mysterious indeed, I can only conclude that the mappings cache that maps local IDs to server IDs is somehow corrupt.

If you make sure you have a backup, you can try triggering a sync from scratch in the profile settings, that should hopefully fix things.

**Reply by Sar6e** [reply]
Thank you for your assistance.

I have a nightly backup in place.  For each of my machines and browsers I was able to disable the failsafe, perform a sync and then enable the failsafe.  Everything is now working as expected.

I love being able to host my own bookmarks in Nextcloud.  The combination with the Floccus extension to syn all my browsers to Nextcloud is so useful to me.  Thank you for your work!
